packages/cadl-ranch-specs/http/payload/json-merge-patch/mockapi.ts

```diff
@@ -79,27 +72,6 @@ Scenarios.Payload_JsonMergePatch_updateResource = passOnSuccess({
       },
     }),
   },
-  handler: (req: MockRequest) => {
-    req.expect.deepEqual(req.body.description, expectedUpdateBody.description);
-    req.expect.deepEqual(req.body.map.key.description, expectedUpdateBody.map.key.description);
-    req.expect.deepEqual(req.body.map.key2, expectedUpdateBody.map.key2);
-    req.expect.deepEqual(req.body.array, expectedUpdateBody.array);
-    req.expect.deepEqual(req.body.intValue, expectedUpdateBody.intValue);
-    req.expect.deepEqual(req.body.floatValue, expectedUpdateBody.floatValue);
-    req.expect.deepEqual(req.body.innerModel, expectedUpdateBody.innerModel);
-    req.expect.deepEqual(req.body.intArray, expectedUpdateBody.intArray);
-    return {
-      status: 200,
-      body: json({
-        name: "Madge",
-        map: {
-          key: {
-            name: "InnerMadge",
-          },
-        },
-      }),
-    };
-  },
```

Comment:
We may need to keep the handler on json-merge-patch (or at least modify the `expectedUpdateBody`.

The expected request may more accurately described in scenarioDoc
https://github.com/Azure/cadl-ranch/blob/aef9ccc66392cc6567c3f7621d99e6217f8ee2b0/packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp#L107-L120
One difference is the `name: "InnerMadge",` part. Since json-merge-patch did not update this field, it can be omitted in request.

Reply:
Omitted `name: "InnerMadge"` from json-merge-patch.